Bio/Entrez/__init__.py

```diff
@@ -188,14 +188,16 @@ def efetch(db, **keywords):
         pass
     else:
         try:
-            ids = ",".join(ids)
-        except TypeError as e:
-            if isinstance(ids, int):
-                ids = str(ids)
-            else:
-                raise e
-        variables["id"] = ids
+            # ids is a single integer or a string representing a single integer
+            ids = str(int(ids))
```

Comment:
Do any of the NCBI databases take non-numeric IDs? i.e. strings where that would fail?

Reply:
Yes, ID can, for example, be a GenBank accession number. There are some examples in the docstring and in the tests which use IDs like "AY851612".

Reply:
They can also have version numbers, see https://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.id_2

Reply:
Thanks - that then falls to the ValueError handler, all fine 👍